python/cog/server/http.py

```diff
@@ -104,6 +104,52 @@ def healthcheck() -> Any:
                 "setup": app.state.setup_result_payload,
             }
         )
+    
+    @app.get("/health/ready")
+    def healthcheck_readiness() -> Any:
+        _check_setup_result()
+        health = app.state.health
+        
+        if health == Health.UNKNOWN:
+            return JSONResponse(
+                {"detail": "Unknown server status"}, status_code=418
+            )
+        if health == Health.SETUP_FAILED:
+            return JSONResponse(
+                {"detail": "Error starting server"}, status_code=500
+            )
+        if health == Health.STARTING:
+            return JSONResponse(
+                {"detail": "Server is starting"}, status_code=503
+            )
```

Comment:
An important note here is that even if the ready probe is failing, the live probe should not (but if we want live probe to wait, we should configure it): https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes